File: Projects/TransportathonHackathon.Clients/angular-ui/src/app/models/response-models/vehicles/getByCompanyIdVehicleResponse.ts
import { Car } from '../../domain-models/car';
import { Driver } from '../../domain-models/driver';
import { PickupTruck } from '../../domain-models/pickupTruck';
import { Truck } from '../../domain-models/truck';

export interface GetByCompanyIdVehicleResponse {
  id: string;
  companyId: string;
  driverId: string;
<<<<<<< Updated upstream
  brand: string;
  model: string;
  modelYear: number;
  createdDate: string;
  updatedDate: string | null;
  deletedDate: string | null;
=======
  createdDate: string;
  updatedDate: string | null;
  deletedDate: string | null;
  brand: string;
  model: string;
  modelYear: number;
>>>>>>> Stashed changes
  driver: Driver | null;
  car: Car | null;
  truck: Truck | null;
  pickupTruck: PickupTruck | null;
}

File: Projects/TransportathonHackathon.API/TransportathonHackathon.Application/Features/Vehicles/Queries/GetByCompanyIdVehicle/GetByCompanyIdVehicleResponse.cs
using TransportathonHackathon.Domain.Entities;

namespace TransportathonHackathon.Application.Features.Vehicles.Queries.GetByCompanyIdVehicle
{
    public class GetByCompanyIdVehicleResponse
    {
        public Guid Id { get; set; }
        public Guid CompanyId { get; set; }
        public Guid DriverId { get; set; }
        public string Brand { get; set; }
        public string Model { get; set; }
        public int ModelYear { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime? UpdatedDate { get; set; }
        public DateTime? DeletedDate { get; set; }
        public virtual Driver? Driver { get; set; }
        public virtual Car? Car { get; set; }
        public virtual Truck? Truck { get; set; }
        public virtual PickupTruck? PickupTruck { get; set; }
    }
}
